src/main/java/de/azapps/kafkabackup/common/offset/OffsetSink.java

```diff
@@ -80,28 +80,26 @@ public void syncOffsets() {
     private void syncOffsetsForGroup(String consumerGroup) throws IOException {
         Map<TopicPartition, OffsetAndMetadata> partitionOffsetsAndMetadata = new HashMap<>();
         partitionsReadLock.lock();
+
         try {
-            // Due to https://issues.apache.org/jira/browse/KAFKA-9507, we need to fetch offsets one TopicPartition at a time
-            // and catch the exception that occurs if the consumer group is missing from the TopicPartition's committed offsets.
-            // TODO: fix when 2.4.1 is released!
-            for (TopicPartition tp : partitions) {
-                ListConsumerGroupOffsetsOptions listConsumerGroupOffsetsOptions = new ListConsumerGroupOffsetsOptions();
-                listConsumerGroupOffsetsOptions.topicPartitions(Collections.singletonList(tp));
-                try {
-                    Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata = adminClient
-                            .listConsumerGroupOffsets(consumerGroup, listConsumerGroupOffsetsOptions)
-                            .partitionsToOffsetAndMetadata()
-                            .get();
-                    partitionOffsetsAndMetadata.put(tp, offsetsAndMetadata.get(tp));
-                } catch (ExecutionException e) {
-                    log.debug("No committed offsets for consumer group {} on topic {} partition {}", consumerGroup, tp.topic(), tp.partition());
-                } catch (InterruptedException e) {
-                    throw new RetriableException(e);
-                }
-            }
+            ListConsumerGroupOffsetsOptions listConsumerGroupOffsetsOptions = new ListConsumerGroupOffsetsOptions();
+            listConsumerGroupOffsetsOptions.topicPartitions((List<TopicPartition>)partitions);
+
+            partitionOffsetsAndMetadata = adminClient
+                .listConsumerGroupOffsets(consumerGroup, listConsumerGroupOffsetsOptions)
+                .partitionsToOffsetAndMetadata()
+                .get()
+                .entrySet()
+                .stream()
+                .filter(map -> map.getValue() != null)
+                .collect(Collectors.toMap(map -> map.getKey(), map -> map.getValue()));
+
+        } catch (InterruptedException | ExecutionException e) {
```

Comment:
Can't find much except this:

> Exception thrown when attempting to retrieve the result of a task that aborted by throwing an exception. This exception can be inspected using the getCause() method.

I changed it to an error thrown instead of log because if we get the exception we will have no data to save. Do you think that make sense?